src/util.ts

```diff
@@ -1200,6 +1200,23 @@ export class Util {
         }
         await fsExtra.writeFile(`${stagingDir}/source/bslib.brs`, source);
     }
+
+    /**
+     * Given a Diagnostic or BsDiagnostic, return a copy of the diagnostic
+     */
+    public toDiagnostic(diagnostic: Diagnostic | BsDiagnostic) {
```

Comment:
the `BsDiagnostic` objects contain a `file` reference which has a `program`, which has an array of files (including the one we just referenced). This causes a recursive error during JSON.stringify when the language server sends the code actions or diagnostics to the client. This function will clone a diagnostic, but only keep the officially supported properties. Thus preventing the JSON.stringify crash.

Reply:
any time we send diagnostics to the client, we should run them through this function.